rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -929,24 +927,30 @@ object ProcNormalizeMatcher {
         // This sorts the None's first, and the uris by lexicographical order.
         // We do this here because the sorting affects the numbering of variables inside the body.
         val sortBindings = newTaggedBindings.sortBy(row => row._1)
-        val newBindings = sortBindings.map { row =>
-          (row._2, row._3, row._4, row._5)
-        }
-        val uris     = sortBindings.flatMap(row => row._1)
-        val newEnv   = input.env.newBindings(newBindings.toList)
-        val newCount = newEnv.count - input.env.count
-        normalizeMatch[M](p.proc_, ProcVisitInputs(VectorPar(), newEnv, input.knownFree))
-          .map { bodyResult =>
-            val resultNew =
-              New(
-                newCount,
-                bodyResult.par,
-                uris,
-                bodyResult.par.locallyFree.from(newCount).map(x => x - newCount)
-              )
-            ProcVisitOutputs(input.par.prepend(resultNew), bodyResult.knownFree)
+        val newBindings  = sortBindings.map(row => (row._2, row._3, row._4, row._5))
+        val uris         = sortBindings.flatMap(row => row._1)
+        val newEnv       = input.env.newBindings(newBindings.toList)
+        val newCount     = newEnv.count - input.env.count
+        val hasAuth      = uris.contains("rho:deployer:auth")
+        if (hasAuth && deployerPk.isEmpty)
+          NormalizerError("No deployer public key set").raiseError[M, ProcVisitOutputs]
+        else {
+          val optDeployerAuth =
+            if (hasAuth) deployerPk.map(pk => DeployerAuth(pk.bytes.toByteString))
```

Comment:
:ok_hand: so my just-in-case comment was not needed :D

I'd though `s/hasAuth/requiresDeployerId`

Reply:
Done